src/mlpack/core/math/lin_alg.cpp

```diff
@@ -85,8 +85,16 @@ void mlpack::math::WhitenUsingEig(const arma::mat& x,
   arma::vec eigenvalues;
 
   // Get eigenvectors of covariance of input matrix.
-  eig_sym(eigenvalues, eigenvectors, ccov(x));
+  arma::mat sym_x = arma::symmatu(ccov(x));
 
+  // Check if Eigen Value exists, if not raise an error log.
+  if (sym_x.is_empty())
+  {
+    Log::Warn << "Eigen Decompositon failed as Eigen Value does not exists ."
+          << std::endl;
+    return;
+  }
+  arma::eig_sym(eigenvalues, eigenvectors, sym_x);
```

Comment:
where is the check for eig_sym() failure?  
hint: it returns a bool set to false if it fails.  
documentation: http://arma.sourceforge.net/docs.html#eig_sym